5333/src/main/cpp/Lift.cpp

```diff
@@ -6,23 +6,6 @@ void LiftManualStrategy::OnUpdate(double dt) {
   _lift.SetManual(speed);
 }
 
-void LiftPresetStrategy::OnUpdate(double dt) {
-  if (_joy.GetRawButton(ControlMap::goalGround)) {
-    _lift.SetSetpoint(ControlMap::setpointGround);
-
-  } else if (_joy.GetRawButton(ControlMap::goalLower1)) {
-    _lift.SetSetpoint(ControlMap::setpointLower1);
-  } else if (_joy.GetRawButton(ControlMap::goalLower2)) {
-  _lift.SetSetpoint(ControlMap::setpointLower2);
-
-  } else if (_joy.GetRawButton(ControlMap::goalMiddle1)) {
-    _lift.SetSetpoint(ControlMap::setpointMiddle1);
-  } else if (_joy.GetRawButton(ControlMap::goalMiddle2)) {
-    _lift.SetSetpoint(ControlMap::setpointMiddle2);
-
-  } else if (_joy.GetRawButton(ControlMap::goalUpper1)) {
-    _lift.SetSetpoint(ControlMap::setpointUpper1);
-  } else if (_joy.GetRawButton(ControlMap::goalUpper2)) {
-    _lift.SetSetpoint(ControlMap::setpointUpper2);
-  }
+void LiftGotoStrategy::OnUpdate(double dt) {
+  _lift.SetSetpoint(_setpoint);
```

Comment:
Do this in OnStart, the setpoint should only be set once. OnUpdate should only really check if the strategy is done (the error is close to 0 and stable)